app/src/main/java/com/lorenzovainigli/foodexpirationdates/view/composable/DateFormatDialog.kt

```diff
@@ -88,30 +110,65 @@ const val DateFormatRow = "DateFormatRow"
 @Composable
 fun DateFormatRow(
     item: String,
+    isSelected: Boolean = false,
     onDismissRequest: () -> Unit,
     onClick: (Context, String) -> Unit
-){
+) {
     val context = LocalContext.current
     val sdf = SimpleDateFormat(item, context.resources.configuration.locales[0])
-    ClickableText(
-        modifier = Modifier.padding(2.dp).testTag(DateFormatRow),
-        text = AnnotatedString(sdf.format(Calendar.getInstance().time)),
-        style = MaterialTheme.typography.headlineSmall.copy(color = MaterialTheme.colorScheme.onSurface),
-        onClick = {
-            onClick(context, item)
-            onDismissRequest()
+    Row(
+        modifier = Modifier
+            .fillMaxWidth()
+            .clip(RoundedCornerShape(8.dp))
+            .background(
+                if (isSelected)
+                    MaterialTheme.colorScheme.primaryContainer.copy(alpha = 0.5f)
+                else
+                    MaterialTheme.colorScheme.surface
+            )
+            .clickable {
+                onClick(context, item)
+                onDismissRequest()
+            }
+            .padding(vertical = 12.dp, horizontal = 8.dp)
+            .testTag(DateFormatRow),
+        verticalAlignment = Alignment.CenterVertically
+    ) {
+        RadioButton(
+            selected = isSelected,
+            onClick = {
+                onClick(context, item)
+                onDismissRequest()
+            }
+        )
+        Spacer(modifier = Modifier.width(12.dp))
+        Text(
+            text = sdf.format(Calendar.getInstance().time),
+            style = MaterialTheme.typography.bodyLarge,
+            color = if (isSelected)
+                MaterialTheme.colorScheme.onPrimaryContainer
+            else
+                MaterialTheme.colorScheme.onSurface,
+            modifier = Modifier.weight(1f)
+        )
+        if (isSelected) {
+            Icon(
+                imageVector = Icons.Default.Check,
+                contentDescription = "Selected",
+                tint = MaterialTheme.colorScheme.primary
+            )
         }
```

Comment:
Do you think this icon is necessary? I think the radio button is sufficient to provide the user with good UI feedback.